### ISSUE TITLE
Applicability of from_cellranger codes on 10X Visium ST-Seq bams

### ISSUE THREAD

**Issue by Prakrithi-P** [fw262/TAR-scRNA-seq#11]
Hey Michael,
I would like to know if the from_cellranger codes can be applied to 10X Visium ST-Seq bams using the correponding refrence datasets. I anticipate it wold work, since the chemistries are quiet similar to that of 10X chromium V3. Just wanted to make sure.

Thanks,
Prakrithi

**Comment by Prakrithi-P** [comment]
Hi Michael, 
I was able to generate TAR_expression_bc_matrix (features.tsv,barcodes.tsv,matrix.mtx) . However I do not know how to proceed with this to plot (as in figure 4).

Where/how  do we get the "Spatial/outs" from these? You have used this directory as input in the Rscript for figure4. Can you guide me through this?

**Comment by fw262** [comment]
Hi Prakrithi,

The "Spatial/outs" folder should look something like this... 
![image](https://user-images.githubusercontent.com/56937181/140956657-cd0edbc4-a21b-4f92-b300-0ba15ce47cf3.png)

It needs to contain the feature_bc_matrices, the `filtered_feature_bc_matrix.h5` file, and a folder called `spatial` which should contain image data for the spatial experiements.

Best,
Michael

**Comment by Prakrithi-P** [comment]
Hi Michael,
     So the TARs which overlap even with the opposite strands of existing annotations are considered aTARs.
With data from probe capture of ST-Seq I find a way high number of uTARs (~16k with bedtools intersect -a TARs -b gene_annotations -s -v) when I consider overlap in a strand specific manner. When I do it only in a positional manner (without considering the strand of the overlap with existing annotations like you have described in the paper -  bedtools intersect -a -b -v ), I have 100s of uTARs.

Since this is a probe based capture protocol, we should not be expecting many uTARs. I was wondering what could be the reason for such huge difference when I consider the overlap with respect to the strand.
-I wanted to understand if the TARs overlapping with the opposite strand of actual gene annotations could actually be considered as aTARs because different genes could actually arise from the same positions but from opp. strands.

Thanks,
Prakrithi

**Comment by fw262** [comment]
Hi Prakrithi,

We do in fact find many more uTARs when we consider strand specificity. It seems that in RNA sequencing data, there will be many reads that align anti-sense to a gene. A cluster of these anti-sense reads would then form a TAR. If we consider directionality, these TARs would be labeled as uTARs. Conversely, if we don't consider directionality, these TARs would be labeled as aTARs since they overlap with an annotated region without considering strandedness. 

It is up to you whether you consider anti-sense gene transcriptional activity as annotated or unannotated. Divergent transcription may explain some of these anti-sense transcriptional behavior. Either way, the uTAR pipeline provides a method to find these potentially interesting transcriptionally active regions.

Best,
Michael

**Comment by Prakrithi-P** [comment]
Thank you so much Michael.
Yes, I guess 16k uTARs with directionality considered is a huge number considering the probe set used is of 18k genes. Guess we'll not consider it.

Thanks,
Prakrithi

**Comment by fw262** [comment]
Yes, that is probably a good idea. Keep in mind that many of these directional "uTARs" will overlap with the same gene.

Michael

**Comment by Prakrithi-P** [comment]
Hi Michael,
I have trouble understanding the value following the strand info in the uTAR nomenclature.

For example, chr7:74195249-74195399_-_33391_0 -> I was thinking the value '33391' here meant coverage. But the read coverage is 3001 calculated with samtools depth and visualized with IGV. This was the case for all the uTARs I checked. Can you please explain how the following command works and what exactly does 'counts' mean here (Column 7 of the bedtools coverage output) ?

bedtools coverage -a ${f}_merge${MERGEBP}.sorted.bed -b <(zcat ${PREFIX}_split.sorted.bed.gz) -s -counts -split -sorted > ${f}_merge${MERGEBP}.sorted.bed_count

Thanks and regards,
Prakrithi

**Comment by fw262** [comment]
The output of `samtools depth` is different from `bedtools coverage` but they should correlate. I believe `samtools depth` skips alignments marked as duplicates, QC fail, etc. while `bedtools coverage` keeps all the reads. Bedtools coverage takes in a list of regions (`-a` input) and a list of sequence alignments (`-b` input) to calculate coverage in each region. Therefore, count means number of reads that fall within each region regardless of alignment quality and duplicates. More information here (https://bedtools.readthedocs.io/en/latest/content/tools/coverage.html).

Michael

**Comment by Prakrithi-P** [comment]
Hi Michael,
   I have been working wit ST-seq datasets (from_cellranger pipeline). I noticed that TAR_expression_matrix_noDir.txt.gz is returned empty for all the datasets I tested while TAR_expression_matrix_withDir.txt.gz gives results. It is fine to just use 'bedtools intersect -a -b -s -v ' on this to annotate the TARs at uTARs without considering the directionality, right?  But still, I would like to understand why an empty file is returned for noDir.

Thanks,
Prakrithi

**Comment by fw262** [comment]
Hi Prakrithi,

It is strange that "noDir" results are empty while you have results with "withDir". For now, you can use the bedtools intersect command to pull uTARs without directionality. Are you getting directional uTARs in all the expected chromosomes/scaffolds?

Best,
Michael

**Comment by Prakrithi-P** [comment]
Hi Michael,
 Yes, I do get directions uTARs in all expected chromosomes.
